src/org/freenetproject/freemail/ui/web/InboxToadlet.java

```diff
@@ -348,16 +348,11 @@ private SortField(String name) {
 		}
 
 		public static SortField fromString(String field) {
-			if("subject".equals(field)) {
-				return SUBJECT;
-			}
-			if("from".equals(field)) {
-				return FROM;
-			}
-			if("date".equals(field)) {
-				return DATE;
+			try {
+				return SortField.fromString(field.toUpperCase());
```

Comment:
Now this is recursion and StackOverflow. It meant `SortField.valueOf(field.toUpperCase())`